### DISCUSSION TITLE
VectorPostprocessor output as a boundary condition

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#23582] [Q&A General] [answered]
Hello,

Please how can I use a Vectorpostprocessor such as the LineValueSampler and SideValueSampler as a Boundary condition without having to output in a csv file. This is because the values in the vectorpostprocessor change at every iteration and I need to update the changed values as new BC at each time step.

Thank you!

**Comment by GiudGiud** [comment]
Hello

Are you using a MultiApp?
Or is the coupling between the two equations (one that computes the VPP, and one that uses) within in the same input file?

Guillaume

**Reply by salaudeen-ya** [reply]
I am using MultiApp with two input files.

The two files compute the VPP separately, and each of them uses the scaled VPP values coming from the other as BC.

**Reply by GiudGiud** [reply]
ok you can transfer the VPP directly between the two applications using this
https://mooseframework.inl.gov/source/transfers/MultiAppVectorPostprocessorTransfer.html

**Reply by salaudeen-ya** [reply]
Alright. Thank you.

**Reply by salaudeen-ya** [reply]
I got this error:
![image](https://user-images.githubusercontent.com/38023107/222494262-b84c5fb5-bbf9-4776-88d1-3fcd811e8c90.png)

This is quite understandable since I only have one subdomain.

To be more clear in my problem:

![image](https://user-images.githubusercontent.com/38023107/222495206-1d8c17b0-8790-41ac-ad73-77a59e5356d2.png)

I want to transfer across the interface using two different input meshes. The VectorPostprocessor only allows me to transfer a scalar value to multiple subdomains at the same time. But, I want to transfer vector of values at the boundary to the other domain. MatchedNode did not quite give me the expected values too.

**Reply by GiudGiud** [reply]
hello

my bad I got this wrong
You have to use this
https://mooseframework.inl.gov/source/transfers/MultiAppReporterTransfer.html#vector_transfer

Guillaume

**Comment by salaudeen-ya** [comment, marked as answer]
Thank you

I was able to effectively transfer using NearestNode and coupled BC
